### PR TITLE
Secondary vertexing for forward rapidities:

### PR DESCRIPTION
**Pull request by sadekr**
- Added FwdDCAFitter tool for fwd tracks secondary vertexing.
  - Helix Helper adapted for both Fwd and central tracks.
  - Some functions added on TrackFwd to accomodate changes.

### CODE REVIEW

**Review by shahor02 — DISMISSED** [review]
Conditionally approving just to trigger CI, don't merge

File: DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackFwd.h
Comment (aphecetche) [inline DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackFwd.h:60]: Hi Rita,

Thanks for the PR. I have some small comments. For the bulk of the PR (`FwdDCAFitterN.h`)  I'll let more knowledgable people comment on...  

Concerning `getSnp` and `getCsp2` I'd have a couple of remarks :

- given that they use data (phi) that are publicly available (getPhi) I don't see the point of making those member functions (I would see the point if they were internally cached or something, but that's not the case here)  

- why use such cryptic names ? Might be obvious to you, but honestly when I see `getSnp` I don't immediately understand `getSinPhi` ;-)  Would matter less in an implementation file, but here we are at the interface level.

File: DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackFwd.h
Comment (aphecetche) [inline DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackFwd.h:122]: A comment for this method would be welcome : what is this "circle" ? (and looking at the implementation I see two cases, maybe worth nothing them briefly in the comment if that's relevant) ?

File: Detectors/Vertexing/src/FwdDCAFitterN.cxx
Comment (aphecetche) [inline Detectors/Vertexing/src/FwdDCAFitterN.cxx:23]: what is this one exactly ?

File: DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackFwd.h
Comment (rpezzi) [reply DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackFwd.h:60]: Hi Laurent, these methods and names were added to keep symmetry with [central barrel track classes](https://github.com/AliceO2Group/AliceO2/blob/dev/DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackParametrization.h#L150-L160) and allow their use in template classes, such as HelixHelper. This is not the (yet) the case for `getSnp` and `getCsp2`. Rita could rename/remove these two in this PR, but this would only increase entropy by diverging from central barrel, unless this is also renamed in other classes in ReconstructionDataFormat.

**Review by aphecetche — APPROVED** [review]

File: Detectors/Vertexing/src/FwdDCAFitterN.cxx
Comment (shahor02) [reply Detectors/Vertexing/src/FwdDCAFitterN.cxx:23]: To trigger templates specialization.

File: Detectors/Vertexing/include/DetectorsVertexing/FwdDCAFitterN.h
Comment (shahor02) [inline Detectors/Vertexing/include/DetectorsVertexing/FwdDCAFitterN.h:311]: @sadekr If I understood correctly, you are seeding in the same way as for the barrel DCAFitter: taking the intersection of 2 circles in X,Y. Not should this is reliable: given a very small lever arm and large errors on helix parameters, such a seed at best will be far from the final convergence point (while the errors are fixed at the seeding point), at worst will not converge. I think for the seeding you should use the point of closest approach in Z in the straight line approximation

Comment (sadekr) [reply Detectors/Vertexing/include/DetectorsVertexing/FwdDCAFitterN.h:311]: Dear Ruben,
As you mentioned, we can not rely on information given from central barrel in forward rapidities. 
This is the reason to the developments of the different seed finder methods. 
As for the HelixHelper,  I only use it as an estimation. 
So it gives us information on the nCrossing (that of course in forward will be mostly 1) 
And also an estimation on X and Y, but that is only used for the conditions. For the actual secondary vertexing; 
the seed finder is used to find a Z (which we can then use to find X Y for each track), and for the time I use the dichotomy method (findZatXY_mid).

File: Detectors/Vertexing/include/DetectorsVertexing/FwdDCAFitterN.h
Comment (shahor02) [inline Detectors/Vertexing/include/DetectorsVertexing/FwdDCAFitterN.h:339]: Hi @sadekr OK, just noticed an independent seeding in `findZatXY_mid` (did not check details), but then why helical crossing and the associated rejections are used? The helical estimate with just a few cm level arm is simply not reliable, it is risky to take any decisions based on it.

**Review by shahor02 — APPROVED** [review]
Hi @sadekr 
Looks fine. One should also add the test which you already have, but this can be done in the next PR.

**Pull request merged**
